Source/Urho3D/Graphics/DebugRenderer.cpp

```diff
@@ -505,7 +541,7 @@ void DebugRenderer::Render()
     graphics->SetBlendMode(lineAntiAlias_ ? BLEND_ALPHA : BLEND_REPLACE);
     graphics->SetColorWrite(true);
     graphics->SetCullMode(CULL_NONE);
-    graphics->SetDepthWrite(true);

```

Comment:
If there's a lot of overlapping lines, it's true the depth writing is a rather important part of debug rendering. Some applications will choose to render debug without depth *testing* in which case it doesn't matter, but it's best to not assume that. Probably the ideal would be to disable depth write during rendering of the filled geometry, and enable otherwise.